src/resspect/database.py

```diff
@@ -1622,16 +1622,17 @@ def save_metrics(self, loop: int, output_metrics_file: str, epoch: int, batch=1)
 
         # write to file)
         queried_sample = np.array(self.queried_sample)
-        flag = queried_sample[:,0].astype(int) == epoch
+        if len(queried_sample) > 0:
```

Comment:
this is part of a small bug fix that I'm not totally sure about and would like a second opinion from one of the science team. When I was setting up the tests for the `learn_loop` function, I ran into an issue with the test data where `queried_sample` was empty, which caused the above line to fail. I added a check here and another place in `database.py` to check for an empty list before continuing, which seems ok in tandem with the `sum(flag) > 0` check before writing, but I wanted to make sure I wasn't causing unexpected behavior.